docs/en/latest/plugin-develop.md

```diff
@@ -437,7 +246,119 @@ function _M.access(conf, ctx)
+to introduce your hook.
+
+Assumed your configuration is:
+
+```yaml
+apisix:
+    ...
+    extra_lua_path: "/path/to/example/?.lua"
+    lua_module_hook: "my_hook"
+```
+
+The `example/my_hook.lua` will be loaded when APISIX starts, and you can use this hook to replace a method in APISIX.
+The example of [my_hook.lua](https://github.com/apache/apisix/blob/master/example/my_hook.lua) can be found under the `example` directory of this project.
+
+## Check external dependencies
+
+If you have dependencies on external libraries, check the dependent items. if your plugin needs to use shared memory, it
+needs to declare via [customizing Nginx configuration](./customize-nginx-configuration.md), for example :
+
+```yaml
+# put this in config.yaml:
+nginx_config:
+    http_configuration_snippet: |
+        # for openid-connect plugin
+        lua_shared_dict discovery             1m; # cache for discovery metadata documents
+        lua_shared_dict jwks                  1m; # cache for JWKs
+        lua_shared_dict introspection        10m; # cache for JWT verification results
+```
+
+The plugin itself provides the init method. It is convenient for plugins to perform some initialization after
+the plugin is loaded. If you need to clean up the initialization, you can put it in the corresponding destroy method.
+
+Note : if the dependency of some plugin needs to be initialized when Nginx start, you may need to add logic to the initialization
+method "http_init" in the file __apisix/init.lua__, and you may need to add some processing on generated part of Nginx
+configuration file in __apisix/cli/ngx_tpl.lua__ file. But it is easy to have an impact on the overall situation according to the
+existing plugin mechanism, **we do not recommend this unless you have a complete grasp of the code**.
+
+## Encrypted storage fields
+
+::: note
+APISIX version is not less than 3.1
+:::
+
+Some plugins require parameters to be stored encrypted, such as the `password` parameter of the `basic-auth` plugin. This plugin needs to specify in the `schema` which parameters need to be stored encrypted.
+
+```lua
+encrypt_fields = {"password"}
+```
+
+If it is a nested parameter, such as the `clickhouse.password` parameter of the `error-log-logger` plugin, it needs to be separated by `.`:
+
+```lua
+encrypt_fields = {"clickhouse.password"}
+```
+
+Currently not supported yet:
+
+1. more than two levels of nesting
+2. fields in arrays
+
+Parameters can be stored encrypted by specifying `encrypt_fields = {"password"}` in the `schema`. APISIX will provide the following functionality.
+
+- When adding and updating resources via the `Admin API`, APISIX automatically encrypts the parameters declared in `encrypt_fields` and stores them in etcd
+- When fetching resources via the `Admin API` and when running the plugin, APISIX automatically decrypts the parameters declared in `encrypt_fields`
```

Comment:
<img width="194" alt="image" src="https://github.com/user-attachments/assets/fb0afb82-5dfa-451a-9752-e80e4dd53a26" />

No need to put Admin API in line.